apps/emqx_bridge_kafka/src/emqx_bridge_kafka_impl_consumer.erl

```diff
@@ -631,6 +638,31 @@ is_dry_run(ConnectorResId) ->
             string:equal(TestIdStart, ConnectorResId)
     end.
 
+-spec check_client_connectivity(pid()) ->
+    ?status_connected
+    | ?status_disconnected
+    | {?status_disconnected, term()}.
+check_client_connectivity(ClientPid) ->
+    case brod_client:get_metadata_safe(ClientPid, <<"__consumer_offsets">>) of
```

Comment:
The `_safe` call fetches metadata for ALL topics in Kafka which is quite expensive.
maybe call `get_group_coordinator` with a made-up group ID instead.

get_group_coordinator will ensure a metadata connection towards one bootstrap endpoint

Reply:
Since this is an internal metadata topic managed by Kafka itself, we could call the non-`safe` version, since it won't be created.

Reply:
:heavy_check_mark: